### PR TITLE
[Fluent 2 iOS] Navigation bar colors update

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

The NavigationBar's colors were updated to match fluent 2 tokens. 

### Verification

Changes were tested on the demo app.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before_brand | after_brand |
| before_light | after_light |
| before_dark | after_dark |
| before | after |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1068)

### CODE REVIEW

File: ios/FluentUI/Navigation/NavigationBar.swift [hunk 1]
```diff
@@ -62,38 +62,35 @@ open class NavigationBar: UINavigationBar {
...
+                return UIColor(dynamicColor: DynamicColor(light: fluentTheme.aliasTokens.colors[.foregroundOnColor].light, dark: fluentTheme.aliasTokens.colors[.foreground1].dark))
             case .system:
-                return Colors.Navigation.System.title
+                return UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.foreground1])
             }
         }
 
-        func backgroundColor(for window: UIWindow, customColor: UIColor?) -> UIColor {
+        func backgroundColor(fluentTheme: FluentTheme, customColor: UIColor?) -> UIColor {
+            let defaultColor = UIColor(dynamicColor: DynamicColor(light: fluentTheme.aliasTokens.colors[.brandBackground1].light, dark: fluentTheme.aliasTokens.colors[.background3].dark))
```
Comment (huwilkes) [inline ios/FluentUI/Navigation/NavigationBar.swift:84]: Is there ever a time we actually want brandBackground1 to still be brand colored in dark mode, or should the alias token be updated to have a neutral dark mode?

**Comment by harrieshin** [comment]
for this control, and any control related to navigationbar we need to do thorough test on iPad splitmode dark as well. Apple has separate "elevated colors" defined for darkmode and fluent1 used to handle it differently. and it all fits together with popover/drawercontroller etc.

**Comment by laminesm** [comment]
> for this control, and any control related to navigationbar we need to do thorough test on iPad splitmode dark as well. Apple has separate "elevated colors" defined for darkmode and fluent1 used to handle it differently. and it all fits together with popover/drawercontroller etc.

Sounds good. I'll create a task to track it and get on it asap.

**Comment by laminesm** [comment]
@harrieshin I have just updated the PR with before and after pictures of tests on iPad splitmode dark. It looks consistent with fluent 1's splitmode. What exactly should I be looking for?

**Pull request merged**